src/tss2-fapi/api/Fapi_Provision.c

```diff
@@ -932,15 +932,46 @@ Fapi_Provision_Finish(FAPI_CONTEXT *context)
             if ((*capabilityData)->data.tpmProperties.tpmProperty[0].value == VENDOR_INTC) {
                 /* Get INTEL certificate for EK public hash via web */
                 uint8_t *cert_buffer = NULL;
-                size_t cert_size;
+                char *cert_start = NULL, *cert_end = NULL, *cert_bin = NULL;
+                size_t cert_size = 0, buffer_size;
                 TPM2B_PUBLIC public;
                 r = ifapi_get_intl_ek_certificate(context, &pkey->public, &cert_buffer,
-                                                  &cert_size);
+                                                  &buffer_size);
                 goto_if_error_reset_state(r, "Get certificates", error_cleanup);
 
-                r = ifapi_cert_to_pem(cert_buffer, cert_size, &command->pem_cert,
-                                      NULL, &public);
+                LOGBLOB_DEBUG(cert_buffer, buffer_size, "%s", "Certificate");
+                /* Parse certificate data out of XML */
+                cert_start = strstr((char *)cert_buffer, "\"certificate\":\"");
+                if (cert_start == NULL) {
+                    SAFE_FREE(cert_buffer);
+                    goto_error_reset_state(r, TSS2_FAPI_RC_GENERAL_FAILURE,
+                                           "Invalid EK cert data", error_cleanup);
+                }
+                cert_start += 15;
+                cert_end = strstr((char *)cert_buffer, "\"}");
```

Comment:
You could also use libjson-c to decode the certs to be more robust against whitespaces added and removed on this stuff in the future.

Reply:
changed to libjson-c parsing